src/lib/dnssd/minimal_mdns/RecordData.cpp

```diff
@@ -101,6 +102,10 @@ bool ParseARecord(const BytesRange & data, chip::Inet::IPAddress * addr)
     addr->Addr[3] = htonl(chip::Encoding::BigEndian::Get32(data.Start()));
 
     return true;
+#else
+    // IPV4 support is disabled: IPAddress should never get IPv4 values.
+    return false;
```

Comment:
Shouldn't this policy-level enforcement happen at a higher layer? Seems like we should be minimizing policy-logic application at this layer since it requires coordinated changes to a fairly broad range of platform impls, when doing so at a higher level would provide more flexibility in how easily we can make such decisions.

I'd think the mDNS impls here should just be responsible for returning to you the set of service records they discovered, and leave it to impls of `CommissioningResolveDelegate::OnNodeDiscovered` to actually prune out the IPv4 records as needed. 

Thoughts?

Reply:
The broader theme for me is to endeavor to keep our expectations of the mDNS logic layers minimal if we can, and keep the corresponding impls as simple and light-weight as possible.

Reply:
I am trying to enforce the invariant of `IPAddress is always a valid IP address` which seems to be the intent as when IPV4 is not enabled, we have:

```
enum class IPAddressType : uint8_t
{
    kUnknown = 0, ///< Not used.
#if INET_CONFIG_ENABLE_IPV4
    kIPv4 = 1, ///< Internet protocol version 4.
#endif         // INET_CONFIG_ENABLE_IPV4
    kIPv6 = 2, ///< Internet protocol version 6.
    kAny  = 3  ///< The unspecified internet address (independent of protocol version).
};
```

The issue here is that `parsing` was done outside the IPAddress class and is reaching into the IP address internals.

Reply:
Bug is that ParseARecord claims to parse a valid IPAddress, but since our Type() does not support IPv4, we claim a totally invalid IPv6 instead:

```
IPAddressType IPAddress::Type() const
{
    if (Addr[0] == 0 && Addr[1] == 0 && Addr[2] == 0 && Addr[3] == 0)
        return IPAddressType::kAny;
#if INET_CONFIG_ENABLE_IPV4
    if (Addr[0] == 0 && Addr[1] == 0 && Addr[2] == htonl(0xFFFF))
        return IPAddressType::kIPv4;
#endif // INET_CONFIG_ENABLE_IPV4
    return IPAddressType::kIPv6;
}
```

Reply:
Resolving after offline chat - we can update and have more consistency as a followup.